### PR TITLE
Fixed Maven tests to work in different locales

### LINKED ISSUE #6171: Locale-dependency of Maven tests
Running `mvn test` with the `fr_FR` system locale gives the following test failures:

```
[ERROR] Failures: 
[ERROR]   ImporterUtilitiesTests.testGetOrAllocateColumn:176 expected [Column 1] but found [Colonne 1]
[ERROR]   SeparatorBasedImporterTests.readSimpleData_0Header_1Row:167 expected [Column 1] but found [Colonne 1]
[ERROR]   SeparatorBasedImporterTests.readSimpleData_0Header_1Row:167 expected [Column 1] but found [Colonne 1]
[ERROR]   SeparatorBasedImporterTests.readSimpleData_0Header_1Row:167 expected [Column 1] but found [Colonne 1]
[ERROR]   SeparatorBasedImporterTests.readSimpleData_RowLongerThanHeader:340 expected [Column 4] but found [Colonne 4]
[ERROR]   SeparatorBasedImporterTests.readSimpleData_RowLongerThanHeader:340 expected [Column 4] but found [Colonne 4]
[ERROR]   SeparatorBasedImporterTests.readSimpleData_RowLongerThanHeader:340 expected [Column 4] but found [Colonne 4]
[ERROR]   ReconOperationTests.serializeReconProcess:121 Objects above are not equal as JSON strings.
```

Tests should pass regardless of the system locale.

### PR DESCRIPTION
**Pull request by elebitzero**
Fixes #6171 

Changes proposed in this pull request:
- Made it so the following Maven tests do not depend on locale:
  - `readSimpleData_0Header_1Row()`
  - `readSimpleData_RowLongerThanHeader()`
  - `serializeReconProcess()`
  - `readXlsxAsText()`

### CODE REVIEW

File: main/tests/server/src/com/google/refine/operations/recon/ReconOperationTests.java [hunk 1]
```diff
@@ -82,7 +83,7 @@ public class ReconOperationTests extends RefineTest {
             "           \"columnName\" : \"researcher\",\n" +
             "           \"expression\" : \"forNonBlank(cell.recon.judgment, v, v, if(isNonBlank(value), \\\"(unreconciled)\\\", \\\"(blank)\\\"))\",\n"
             +
-            "           \"name\" : \"researcher: judgment\"\n" +
+            "           \"name\" : \"researcher: " + OpenRefineMessage.recon_operation_judgement_facet_name() + "\"\n" +
```
Comment (wetneb) [inline main/tests/server/src/com/google/refine/operations/recon/ReconOperationTests.java:86]: This needs escaping to make sure the JSON syntax is still valid.

File: main/tests/server/src/com/google/refine/importers/ExcelImporterTests.java [hunk 2]
```diff
@@ -268,8 +268,8 @@ public void readXlsxAsText() throws IOException {
 
         Assert.assertEquals(project.rows.size(), ROWS);
         Assert.assertEquals(project.rows.get(1).cells.size(), COLUMNS);
-        Assert.assertEquals(((String) project.rows.get(1).getCellValue(0)), NUMBER_FORMAT.format(1.1));
-        Assert.assertEquals(((String) project.rows.get(2).getCellValue(0)), NUMBER_FORMAT.format(2.2));
+        Assert.assertEquals((String) project.rows.get(1).getCellValue(0), "1.1");
+        Assert.assertEquals((String) project.rows.get(2).getCellValue(0), "2.2");
```
[the remark this answers is not in the document]
Comment (wetneb) [reply main/tests/server/src/com/google/refine/importers/ExcelImporterTests.java:272]: I ran the tests on this branch, by setting my locale to `fr_FR` ([via Maven's settings](https://stackoverflow.com/a/48244089/985087)) and got the following error:
```
[ERROR] Failures: 
[ERROR]   ExcelImporterTests.readXlsxAsText:271 expected [1.1] but found [1,1]
```

[the remark this answers is not in the document]
Comment (elebitzero) [reply main/tests/server/src/com/google/refine/importers/ExcelImporterTests.java]: I reproduced the problem you see when setting `user.language` and `user.region` to `fr_FR`. I was setting my system locale to France in Windows settings, and it was working.

I debugged the problem and found that `NumberFormat.getNumberInstance()` was returning an incorrect formatter, but if I call `NumberFormat.getNumberInstance(Locale.getDefault())` it works.

File: main/tests/server/src/com/google/refine/operations/recon/ReconOperationTests.java (hunk 1 above)
Comment (elebitzero) [reply main/tests/server/src/com/google/refine/operations/recon/ReconOperationTests.java]: Ok, I escaped the strings with `StringEscapeUtils.escapeJson()`.

**Review by wetneb — APPROVED** [review]

**Comment by wetneb** [comment]
Thanks a lot!

**Pull request merged**